api/src/controllers/admin/deleteRole.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from 'express';
+import { CamelCasedProperties } from 'type-fest';
+import type * as s from 'zapatos/schema';
+
+import * as auth0Repository from '../../repositories/auth0Repository';
+import * as userRolesRepository from '../../repositories/userRolesRepository';
+import Role from '../../types/roles';
+import { decodeQueryToUser, manyToCamelCase } from '../../util/helper';
+import controller from '../controllerUtil';
+import Validator, { beAValidRole, beAValidUser } from '../validation';
+
+type ResBody = {
+    users?: CamelCasedProperties<s.user_roles.JSONSelectable>[];
+    message?: string;
+};
+
+type Params = {
+    userId: string;
+    role: string;
+};
+
+class ParamsValidator extends Validator<Params> {
+    constructor() {
+        super('route parameters');
+
+        this.ruleFor('userId').mustAsync(beAValidUser);
+
+        this.ruleFor('role').mustAsync(beAValidRole);
+    }
+}
+
+const deleteRole = controller(async (req: Request<Params, ResBody>, res: Response<ResBody>): Promise<void> => {
+    await new ParamsValidator().validateAndThrow(req.params);
```

Comment:
`req.params` return the route parameters, from the route in `adminRoutes`, I don't see `userId` or `role` anywhere, did you mean to use `req.body`?

Reply:
Since the route is now updated to `/users/{userId}/role/{role}` we now have route parameters. Sweet1